core/opendaq/signal/include/opendaq/data_packet_impl.h

```diff
@@ -595,20 +602,64 @@ ErrCode DataPacketImpl<TInterface>::getLastValue(IBaseObject** value, ITypeManag
     if (dimensionCount > 1)
         return OPENDAQ_IGNORED;
 
-    void* addr;
-    ErrCode err = this->getData(&addr);
-    if (OPENDAQ_FAILED(err))
-        return err;
-
-    addr = static_cast<char*>(addr) + (sampleCount - 1) * descriptor.getSampleSize();
+    readLock.lock();
```

Comment:
We should use a scoped lock/lock guard here. Otherwise this needs to be wrapped in a try/catch/finally, as this lock will not lock on caught exceptions.